Readme.md

```diff
@@ -38,3 +38,89 @@ In production mode, we don't want on the fly transpilation. Instead we want to o
 - default-babel-rc (server prod time)
 - bin/quintype-build (server prod time)
 - webpack.config.js (client)
+
+### Requirements of new babel/webpack configs
+
+- should allow running only babel on the server code
+- should support BABEL_TARGET and NODE_ENV to allow generating config for browser/server and dev/prod combinations
+- should be able override babel config via app specific JS
+- should be able override webpack config via app specific JS
+- should be able to pass it to jest/mocha with necessary options set
+- should be able to get entire babel/webpack config as a JS object to easily merge it with storybook config
+
+For Jest (tests),
+- Use a custom transformer file to give the path of babel config
+
+For Babel Register (server side, dev mode),
+- It directly accepts the path to the config file
+
+For Babel CLI (server side, prod mode),
+- It directly accepts the path to the config file
+
+For Webpack (browser, prod & dev mode)
+- Babel Loader can be given the path to the babel config file.
+- Babel Loader can also accept the entire babel config as an object
+
+Steps,
+- The babel config file is loaded
+- It will read environment variables BABEL_TARGET and NODE_ENV to determine if config is for server/browser and prod/dev.
+- It will read the `quintype-build.config.js` file if present in the application root, ie the working directory.
+- The file will export build options object. It can have functions like `modifyBabel`, `modifyWebpack` to allow the developer to customize the babel/webpack config.
+- The default config is prepared based on environment variables and then passed to `modifyBabel` for augmentation. The returned config is exported.
+
+### Migration to build@3.0.0
+
+- Remove all packages that have been moved to build or that are not required.
+```sh
+npm uninstall babel-cli babel-core babel-loader babel-plugin-dynamic-import-node babel-plugin-transform-assets-import-to-string babel-preset-env babel-preset-es2015 babel-preset-es2015-tree-shaking babel-preset-react babel-loader duplicate-package-checker-webpack-plugin webpack-common-shake
+```
+
+- Update build
+```sh
+npm install @quintype/build@latest
+or
+npm install @quintype/build@3.0.0
+```
+
+- Update package.json to add the name of the publisher, this will be used for namespacing the assets folder.
+
```

Comment:
👍  I keep forgetting changing the publisher name in webpack config which results in unfound JS and CSS builds.